docs/r2/tensorboard_quickstart.ipynb

```diff
@@ -0,0 +1,463 @@
+        "  for x_train, y_train in dataset:\n",
+        "    loss = train_step(model, optimizer, x_train, y_train)\n",
+        "    avg_loss(loss)\n",
+        "\n",
+        "    if tf.equal(optimizer.iterations % log_freq, 0):\n",
+        "      tf.summary.scalar('loss', avg_loss.result(), step=optimizer.iterations)\n",
+        "      tf.summary.scalar('accuracy', compute_accuracy.result(), step=optimizer.iterations)\n",
+        "      print('Step #%d\\tLoss: %.6f' % (optimizer.iterations, loss))\n",
+        "    avg_loss.reset_states()\n",
+        "    compute_accuracy.reset_states()\n",
+        "\n",
+        "def test(model, dataset, step_num):\n",
+        "  avg_loss = tf.keras.metrics.Mean('loss', dtype=tf.float32)\n",
+        "\n",
+        "  for (x_test, y_test) in dataset:\n",
+        "    logits = model(x_test, training=False)\n",
+        "    avg_loss(compute_loss(y_test, logits))\n",
+        "    compute_accuracy(y_test, logits)\n",
+        "  print('Model test set loss: {:0.4f} accuracy: {:0.2f}%'.format(\n",
+        "      avg_loss.result(), compute_accuracy.result() * 100))\n",
+        "  tf.summary.scalar('loss', avg_loss.result(), step=step_num)\n",
+        "  tf.summary.scalar('accuracy', compute_accuracy.result(), step=step_num)\n",
+        "    "
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "kvbPGxI2hZ4h",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Train and test the model within the scope of summary writers to write the summaries to disk."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "0wxFZp8JDXb1",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "current_time = datetime.now().strftime(\"%Y%m%d-%H%M%S\")\n",
+        "train_log_dir = 'logs/gradient_tape/' + current_time + '/train'\n",
+        "test_log_dir = 'logs/gradient_tape/' + current_time + '/test'\n",
+        "train_summary_writer = tf.summary.create_file_writer(train_log_dir)\n",
+        "test_summary_writer = tf.summary.create_file_writer(test_log_dir)\n",
+        "\n",
+        "model = create_model() # reset our model\n",
+        "\n",
+        "with train_summary_writer.as_default():\n",
+        "  train(model, optimizer, train_dataset)\n",
+        "\n",
+        "with test_summary_writer.as_default():\n",
+        "  test(model, test_dataset, optimizer.iterations)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Fwhsw0zN23wm",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Open TensorBoard in the same way way but point it at a different log directory. After starting TensorBoard, you can rerun the previous cells to start training again and monitor the training within TensorBoard while it's happening.\n",
```

Comment:
It’s a bit funny-looking that there’s a single `test` point in the
resulting plot:

![Screenshot of rendered output after training](https://user-images.githubusercontent.com/4317806/53213933-8e30b480-35ff-11e9-94a5-4a88296a5cc3.png)

Any thoughts?

Reply:
I've updated everything to make the gradient tape scenario have train/test that map to each other and have several epochs.
I based the code on the [advanced quickstart](https://github.com/tensorflow/docs/blob/master/site/en/r2/tutorials/quickstart/quickstart_advanced.ipynb) @yashk2810 added. It might be worth considering also adding the TensorBoard code there at some point.